components/match2/commons/match_group_input.lua

```diff
@@ -589,7 +587,8 @@ function MatchGroupInput.readMvp(match)
 	-- parse the players to get their information
 	local parsedPlayers = Array.map(players, function(player, playerIndex)
 		local link = mw.ext.TeamLiquidIntegration.resolve_redirect(mw.text.split(player, '|')[1]):gsub(' ', '_')
-		for _, opponent in Table.iter.pairsByPrefix(match, 'opponent') do
+		local opponents = MatchGroupUtil.normalizeSubtype(match, 'opponent', match.match2opponents or match.opponents)
```

Comment:
could move this to before the Array.map so we only call it once instead of for each player